banzai_nres/qc/qc_wavelength.py

```diff
@@ -2,48 +2,109 @@
-                        'order_diff_Dlambda': order_diff_Dlambda}
+        Dlambda = self.calculate_delta_lambda(image, lab_lines)
+        result = self.calculate_1d_metrics(image, Dlambda, lab_lines, Dlambda_match_threshold)
+        sigma_Dlambda, matched_sigma_Dlambda, chi2, matched_chi2, num_matched_lines, velocity_precision = result
+        #x_diff_Dlambda, order_diff_Dlambda = self.calculate_2d_metrics(image, Dlambda)
+        # TODO add the number of overlaps matched during wavelength calibration to this metric.
+        #  the number of fit overlaps may be saved in the features header, check this.
+        qc_results = {'sigma_Dlambda': sigma_Dlambda,
+                      'matched_sigma_Dlambda': matched_sigma_Dlambda,
+                      'wavecal_precision(m/s)': velocity_precision.to(units.meter/units.second).value,
+                      'chi_squared': chi2,
+                      'matched_chi_squared': matched_chi2,
+                      'num_matched_lines': num_matched_lines}
+        qc_description = {'sigma_Dlambda': 'Wavecal statistic. Standard deviation of the distribution of '
+                                           'lab minus measured wavelength residuals',
+                          'matched_sigma_Dlambda': 'Wavecal statistic. sigma_Dlambda but only using lines that agree with a '
+                                                   f'lab line to within {Dlambda_match_threshold} Angstroms.',
+                          'wavecal_precision(m/s)': 'matched_sigma_Dlambda/sqrt(N) in velocity space '
+                                                    '(i.e. standard deviation of delta lambda/lambda * c divided by '
+                                                    'the square root of the number of matched lines) '
+                                                    'for matched lines only. I.e. this is the formal error on the mean '
+                                                    'of the residuals distribution (in velocity units). '
+                                                    'Units of meter per second.',
+                          'chi_squared': 'Wavecal statistic. Formal chi-squared statistic of the wavelength residuals '
+                                         'using their formal errors.',
+                          'matched_chi_squared': 'Wavecal statistic. chi_squared but only using lines that agree with a '
+                                                 f'lab line to within {Dlambda_match_threshold} Angstroms.',
+                          'num_matched_lines': f'Wavecal statistic. Number of measured lines that agree with a '
+                          f'lab line to within {Dlambda_match_threshold} Angstroms.'}
         qc.save_qc_results(self.runtime_context, qc_results, image)
+        # saving the results to the image header
+        for key in qc_results.keys():
+            image.meta[key] = (qc_results[key], qc_description[key])
+
```

Comment:
check that this is satisfactory

Reply:
qc_description should go in docs. and then the comment in the fits file should be e.g. 'wavecal precision (m/s)' and 'chi squared of wavelength solution'. Max of 20ish characters.

Change the qc_keys to be <= 8 characters.

Reply:
move descriptions (and shorten them) to docs. Point the reader to the function that calculates this. Code is open source so that would be good.